### PR TITLE
Deprecate the wrap-mode API and add flex-wrap

### LINKED ISSUE #127: Rename setWrapMode to setFlexWrap
The properties `flex-grow`, `flex-shrink`, `flex-direction`, and `flex-basis` all have `flex` prefixes in the corresponding Flow methods, e.g. `setFlexGrow`.

`flex-wrap`, however, does not, as the method is called `setWrapMode`.

This is confusing, as the method does not appear as a suggestion in an IDE when typing `setFlex...`.

### PR DESCRIPTION
**Pull request by hongduc-phan**
Fixes #127

### CODE REVIEW

File: vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java [hunk 1]
```diff
@@ -98,14 +98,28 @@ public FlexLayout(Component... children) {
     }
 
     /**
+     * @deprecated Rename setWrapMode to setFlexWrap
```
Comment (DiegoCardoso) [inline vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java]: Please, change this to something like:
```suggestion
     * @deprecated Use {@link #setFlexWrap(WrapMode)} instead
```

(confirm if `@link` is right, first)

File: vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java [hunk 2]
```diff
@@ -115,14 +129,28 @@ public void setWrapMode(WrapMode wrapMode) {
     }
 
     /**
+     * @deprecated Rename getWrapMode to getFlexWrap
```
Comment (DiegoCardoso) [inline vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java]: Same as previous.

File: vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java [hunk 3]
```diff
@@ -98,14 +98,28 @@ public FlexLayout(Component... children) {
...
+
+    /**
+     * Gets the {@link WrapMode} used by this layout.
+     * <p>
+     * The default flex wrap mode is {@link WrapMode#NOWRAP}.
+     *
+     * @param wrapMode the flex wrap mode of the layout, never
+     *                     <code>null</code>
+     */
+    public void setFlexWrap(WrapMode wrapMode) {
```
Comment (tomivirkki) [inline vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java]: Also deprecate the `WrapMode` enum and create a new one (`FlexWrap`)

File: vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java (hunk 1 above)
Comment (hongduc-phan) [reply vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java]: after using @link, that link is pointing to the right methods. Thank you for comments

File: vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java (hunk 3 above)
Comment (hongduc-phan) [reply vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java]: thank you for comment. It is done.

File: vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java [hunk 4]
```diff
@@ -77,6 +79,44 @@ static WrapMode toWrapMode(String flexValue, WrapMode defaultValue) {
...
+
+        FlexWrap(String flexValue) {
+            this.flexValue = flexValue;
+        }
+
+        String getFlexValue() {
+            return flexValue;
+        }
+
+        static FlexWrap toWrapMode(String flexValue, FlexWrap defaultValue) {
```
Comment (tomivirkki) [inline vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java]: Shouldn't say "mode"

File: vaadin-ordered-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/orderedlayout/it/OrderedLayoutITView.java [hunk 5]
```diff
@@ -40,6 +42,30 @@ public OrderedLayoutITView() {
...
+        noWrap.setId("no-wrap");
+        wrap.setId("wrap");
+        wrapReverse.setId("wrap-reverse");
+        getWrapMode.setId("wrap-mode");
+        flayout.add(
+                noWrap,
+                wrap,
+                wrapReverse,
+                getWrapMode,
+                wrapModeText);
```
Comment (tomivirkki) [inline vaadin-ordered-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/orderedlayout/it/OrderedLayoutITView.java]: Shouldn't say "mode"

File: vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java (hunk 1 above)
Comment (hongduc-phan) [reply vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java]: changed!

File: vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java (hunk 2 above)
Comment (hongduc-phan) [reply vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java]: changed

File: vaadin-ordered-layout-flow-integration-tests/src/test/java/com/vaadin/flow/component/orderedlayout/tests/OrderedLayoutIT.java [hunk 6]
```diff
@@ -24,6 +25,36 @@ public void testVerticalLayout_spacingTheme() {
...
+        btnWrap.click();
+        Assert.assertEquals("wrap",
+                fLayout.getCssValue("flex-wrap"));
+
+        btnWrapReverse.click();
+        Assert.assertEquals("wrap-reverse",
+                fLayout.getCssValue("flex-wrap"));
+
+        btnGetWrapMode.click();
+        Assert.assertEquals("WRAP_REVERSE", wrapModeText.getText());
```
Comment (tomivirkki) [inline vaadin-ordered-layout-flow-integration-tests/src/test/java/com/vaadin/flow/component/orderedlayout/tests/OrderedLayoutIT.java]: Shouldn't say "mode"

File: vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java [hunk 7]
```diff
@@ -115,18 +157,52 @@ public void setWrapMode(WrapMode wrapMode) {
...
+     * <p>
+     * The default flex wrap mode is {@link FlexWrap#NOWRAP}.
+     *
+     * @param flexWrap the flex wrap mode of the layout, never
+     *                     <code>null</code>
+     */
+    public void setFlexWrap(FlexWrap flexWrap) {
+        if (flexWrap == null) {
+            throw new IllegalArgumentException(
+                    "The 'wrapMode' argument can not be null");
```
Comment (tomivirkki) [inline vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java]: Shouldn't say "mode"

File: vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java [hunk 8]
```diff
@@ -115,18 +157,52 @@ public void setWrapMode(WrapMode wrapMode) {
...
+    /**
+     * Gets the current flex wrap mode of the layout.
+     * <p>
+     * The default flex wrap mode is {@link FlexWrap#NOWRAP}.
+     *
+     * @return the flex wrap mode used by the layout, never
+     * <code>null</code>
+     */
+    public FlexWrap getFlexWrap() {
+        return FlexWrap.toWrapMode(
```
Comment (tomivirkki) [inline vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java]: Shouldn't say "mode"

File: vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java (hunk 4 above)
Comment (hongduc-phan) [reply vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java]: thank you for correcting me. I changed.

File: vaadin-ordered-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/orderedlayout/it/OrderedLayoutITView.java (hunk 5 above)
Comment (hongduc-phan) [reply vaadin-ordered-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/orderedlayout/it/OrderedLayoutITView.java]: thank you for correcting me. I changed.

File: vaadin-ordered-layout-flow-integration-tests/src/test/java/com/vaadin/flow/component/orderedlayout/tests/OrderedLayoutIT.java (hunk 6 above)
Comment (hongduc-phan) [reply vaadin-ordered-layout-flow-integration-tests/src/test/java/com/vaadin/flow/component/orderedlayout/tests/OrderedLayoutIT.java]: thank you for correcting me. I changed.

File: vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java (hunk 7 above)
Comment (hongduc-phan) [reply vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java]: thank you for correcting me. I changed.

File: vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java (hunk 8 above)
Comment (hongduc-phan) [reply vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/FlexLayout.java]: thank you for correcting me. I changed.

**Comment by tomivirkki** [comment]
The PR should not base on `master`, but `2.1`. Once `2.1` is stable, we will forward-port the new API's among this deprecation to `master`.

**Comment by hongduc-phan** [comment]
#132

**Comment by tomivirkki** [comment]
You can change the base of a PR by hitting the "Edit" button

**Comment by hongduc-phan** [comment]
https://github.com/vaadin/vaadin-ordered-layout-flow/pull/132

**Pull request closed without merging**